src/tests/blueprint/t_blueprint_mesh_generate_map_fields.cpp

```diff
@@ -0,0 +1,152 @@
+
+    uint32 *id_values = side_mesh["fields/original_element_ids/values"].value();
+    
+    int i = 0;
+    for (int j = 0; j < num_polygons; j ++)
+    {
+        if (j % 2)
+        {
+            EXPECT_EQ(id_values[i], j);
+            EXPECT_EQ(id_values[i + 1], j);
+            EXPECT_EQ(id_values[i + 2], j);
+            EXPECT_EQ(id_values[i + 3], j);
+            i += 4;
+        }
+        else
+        {
+            EXPECT_EQ(id_values[i], j);
+            EXPECT_EQ(id_values[i + 1], j);
+            EXPECT_EQ(id_values[i + 2], j);
+            EXPECT_EQ(id_values[i + 3], j);
+            EXPECT_EQ(id_values[i + 4], j);
+            EXPECT_EQ(id_values[i + 5], j);
+            EXPECT_EQ(id_values[i + 6], j);
+            EXPECT_EQ(id_values[i + 7], j);
+            i += 8;
+        }
+    }
+}
+
+//-----------------------------------------------------------------------------
+TEST(conduit_blueprint_generate_unstructured, generate_sides_and_map_fields_3D)
+{
+    index_t length = 1;
+    Node n, side_mesh, info;
+
+    // create a polychain of length 1
+    examples::polychain(length, n);
+    EXPECT_TRUE(verify(n, info));
+
+    topology::unstructured::generate_sides_and_map_fields(n, side_mesh, "topo");
+    EXPECT_TRUE(verify(side_mesh, info));
+
+    EXPECT_EQ(side_mesh["fields/chain/topology"].as_string(), "topo");
+    EXPECT_EQ(side_mesh["fields/chain/association"].as_string(), "element");
+    EXPECT_EQ(side_mesh["fields/chain/volume_dependent"].as_string(), "false");
+
+    index_t num_tets_in_hex = 24;
+    index_t num_tets_in_triprism = 18;
+
+    index_t num_field_values = num_tets_in_hex + 2 * num_tets_in_triprism;
+    EXPECT_EQ(side_mesh["fields/chain/values"].dtype().number_of_elements(), num_field_values);
+
+    int64 *chain_values = side_mesh["fields/chain/values"].value();
+
+    for (int i = 0; i < num_field_values; i ++)
+    {
+        if (i < num_tets_in_hex)
+        {
+            EXPECT_EQ(chain_values[i], 0);
+        }
+        else
+        {
+            EXPECT_EQ(chain_values[i], 1);
+        }
+    }
+
+    EXPECT_EQ(side_mesh["fields/original_element_ids/topology"].as_string(), "topo");
+    EXPECT_EQ(side_mesh["fields/original_element_ids/association"].as_string(), "element");
+    EXPECT_EQ(side_mesh["fields/original_element_ids/volume_dependent"].as_string(), "false");
+
+    EXPECT_EQ(side_mesh["fields/original_element_ids/values"].dtype().number_of_elements(), num_field_values);
+
+    uint32 *id_values = side_mesh["fields/original_element_ids/values"].value();
+
+    for (int i = 0; i < num_field_values; i ++)
+    {
+        if (i < num_tets_in_hex)
+        {
+            EXPECT_EQ(id_values[i], 0);
+        }
+        else if (i < num_tets_in_hex + num_tets_in_triprism)
+        {
+            EXPECT_EQ(id_values[i], 1);
+        }
+        else
+        {
+            EXPECT_EQ(id_values[i], 2);
+        }
+    }
+}
```

Comment:
I think it would be good to also add separate tests for the the `map_fields_to_generated_sides` methods as well. So they can serve as another example and so test if the calling conventions are different than the all in one.